### PR TITLE
Add DevTask-3

### PR DESCRIPTION
**Pull request by KirillKhripach**

### CODE REVIEW

File: DevTask3/DevTask3/Company.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace DevTask3
+{
+    enum Qualification
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/Company.cs]: please move enum to separate file

File: DevTask3/DevTask3/Company.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+                    employeesAmount[(int)Qualification.Junior]++;
+                    continue;
+                }
+            }
+            Console.WriteLine("The number of employees you need:");
+            Console.WriteLine($"Junior: {employeesAmount[(int)Qualification.Junior]}");
+            Console.WriteLine($"Middle: {employeesAmount[(int)Qualification.Middle]}");
+            Console.WriteLine($"Senior: {employeesAmount[(int)Qualification.Senior]}");
+            Console.WriteLine($"Lead: {employeesAmount[(int)Qualification.Lead]}");
+            return employeesAmount;
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/Company.cs]: please add blank line before return

File: DevTask3/DevTask3/Company.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                Console.WriteLine("The company can provides the required number of employees");
+            }
+        }
+
+        /// <summary>
+        /// Displays optimized list of employees
+        /// </summary>
+        /// <param name="employees">Optimized list of employees</param>
+        /// <returns>Array of amount of employees</returns>
+        public int[] DisplayOptimizedList(List<Employee> employees)
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/Company.cs]: is it better to create property team and name this method like DisplayTeam.
becase for me it is something strange that object Company should  DisplayOptimizedList

File: DevTask3/DevTask3/CriterionChooser.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    class CriterionChooser
+    {
+        /// <summary>
+        /// Chooses criterion according to input data
+        /// </summary>
+        /// <returns>Choosed criterion</returns>
+        public Optimizer Choose()
+        {
+            Console.WriteLine("Select a criterion to calculate:\n1. Maximum productivity\n2. Minimum cost\n3. Minimum number of employees without juniors");
+            int choice = Int32.Parse(Console.ReadLine());
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/CriterionChooser.cs]: if I enter something like 'asdfrtbvwref' ?

File: DevTask3/DevTask3/EntryPoint.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace DevTask3
+{
+    class EntryPoint
+    {
+        /// <summary>
+        /// Entry point
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/EntryPoint.cs]: it is not a good summary. please describe this method

File: DevTask3/DevTask3/EntryPoint.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace DevTask3
+{
+    class EntryPoint
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/EntryPoint.cs]: please add summary

File: DevTask3/DevTask3/FirstCriterionOptimizer.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+                        continue;
+                    }
+                    if (qualification is Junior)
+                    {
+                        necessaryEmployees.Add(new Junior());
+                        continue;
+                    }
+                }
+            }
+            return necessaryEmployees;
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/FirstCriterionOptimizer.cs]: please add blank lines between different block of code

Example:
int number;

foreach ()
{

}

return

File: DevTask3/DevTask3/Middle.cs [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    /// <summary>
+    /// Class for middle type employee
+    /// </summary>
+    class Middle : Junior
+    {
+        public Middle()
+        {
+            Salary = 500;
+            Productivity = 25;
+            Value = (double)Salary / Productivity;
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/Middle.cs]: does the calculation of optimality different for all Employees?

File: DevTask3/DevTask3/SecondCriterionOptimizer.cs [hunk 9]
```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections.Generic;
+
+namespace DevTask3
+{
+    /// <summary>
+    /// Class for optimization
+    /// According to second criterion
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/SecondCriterionOptimizer.cs]: 'second"? i don't know what is it? Please describe this criterion

File: DevTask3/DevTask3/SecondCriterionOptimizer.cs [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    /// </summary>
+    class SecondCriterionOptimizer : ThirdCriterionOptimizer
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/SecondCriterionOptimizer.cs]: Please rename. MaxProductivity, Minsalary etc

File: DevTask3/DevTask3/ThirdCriterionOptimizer.cs [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+            employeeQualifications.Add(new Lead());
+            var sortedEmployeeQualifications = employeeQualifications.OrderBy(i => i.Value);
+            int number;
+            foreach (Employee qualification in sortedEmployeeQualifications)
+            {
+                number = CriterionParameter / qualification.Productivity;
+                CriterionParameter -= qualification.Productivity * number;
+                for (int i = 0; i < number; i++)
+                {
+                    if (qualification is Lead)
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/ThirdCriterionOptimizer.cs]: Is it possible to use else if or switch case without using continue?

File: DevTask3/DevTask3/ThirdCriterionOptimizer.cs [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+                        necessaryEmployees.Add(new Junior());
+                        continue;
+                    }
+                }
+            }
+            if (CriterionParameter > 0)
+            {
+                int minimumSalary = employeeQualifications.Min(Employee => Employee.Salary);
+                Employee qualification = employeeQualifications.First(Employee => Employee.Salary == minimumSalary);
+                if (qualification is Lead)
```
Comment (AlexeiRomankov) [inline DevTask3/DevTask3/ThirdCriterionOptimizer.cs]: duplicated code

File: DevTask3/DevTask3/CriterionChooser.cs (hunk 4 above)
Comment (KirillKhripach) [reply DevTask3/DevTask3/CriterionChooser.cs]: Exception

**Pull request merged**